src/docfx/build/link/LinkResolver.cs

```diff
@@ -157,7 +158,48 @@
             var resolvedHref = _config.RemoveHostName ? UrlUtility.RemoveLeadingHostName(href, _config.HostName) : href;
             resolvedHref = UrlUtility.RemoveLeadingHostName(resolvedHref, _config.AlternativeHostName, true);
 
-            return (errors, resolvedHref, fragment, LinkType.AbsolutePath, null, false);
+            if (href != resolvedHref)
+            {
+                if (TryResolveAbsolutePathAsRelativePath(resolvedHref, hrefRelativeTo, _buildScope.GetContentType(resolvedHref) == ContentType.Resource, true, out var relativePath))
+                {
+                    if (relativePath == null)
+                    {
+                        errors.Add(Errors.Link.FileNotFound(new SourceInfo<string>(href, href)));
+                    }
+                    else
+                    {
+                        href = new SourceInfo<string>(resolvedHref.TrimStart(new char[] { '/', '\\' }).Remove(_config.BasePath.ToString().Length));
+                        file = relativePath;
+                        linkType = LinkType.RelativePath;
```

Comment:
Why don't we return href in this case? Do we require the later process in this method?